### PR TITLE
feat(wallet): Backed Up Transactions Warning

### PR DESCRIPTION
**Pull request by Douglashdaniel**
## Description 
Implemented a Backed Up Transactions Warning

In the case that user has submitted a transaction and the miners are not able to pick it up due to low submitted `gas`.
This warning will show if a user attempts to create another transaction and the old `Submitted` one is over `24` hours old.

1) If user chooses too `Clear and continue` we override the old transaction by setting the `nonce` of the new `Unapproved` transaction to the `nonce` of the old stuck `Submitted` transaction.
2) If the user chooses to not `Clear and continue` we treat the new `Unapproved` transaction as normal.
3) If a user has multiple `Unapproved` transactions in queue, the first transactions will display the warning with the `Clear and continue` option. If chosen the next `Unapproved` transactions will show the same warning with out the `Clear and continue` option until the first transaction is `Confirmed` on the blockchain.

Resolves <https://github.com/brave/brave-browser/issues/21051>

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/handbook/blob/master/development/security.md#when-is-a-security-review-needed), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/handbook/blob/master/development/security.md#when-is-a-security-review-needed), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

- Demo (note: For this demo I set it to show the warning after a `Submitted` transaction is 30 seconds old, will actually show if a transaction is still returned as `Submitted` after 24 hours.)

https://user-images.githubusercontent.com/40611140/156451218-bdf93f24-7e8a-4b48-81d2-bd52dbb85748.mov

### CODE REVIEW

File: components/brave_wallet_ui/components/extension/confirm-transaction-panel/index.tsx [hunk 1]
```diff
@@ -144,11 +157,44 @@ function ConfirmTransactionPanel (props: Props) {
   const [currentTokenAllowance, setCurrentTokenAllowance] = React.useState<string>('')
   const [isEditingAllowance, setIsEditingAllowance] = React.useState<boolean>(false)
   const [showAdvancedTransactionSettings, setShowAdvancedTransactionSettings] = React.useState<boolean>(false)
+  const [showTransactionsBackedUpWarning, setShowTransactionsBackedUpWarning] = React.useState<boolean>(false)
 
   const { findAssetPrice } = usePricing(transactionSpotPrices)
   const parseTransaction = useTransactionParser(selectedNetwork, accounts, transactionSpotPrices, visibleTokens, fullTokenList)
   const transactionDetails = parseTransaction(transactionInfo)
 
+  const calculateDiffInMilliseconds = (date: Date) => {
```
Comment (josheleonard) [inline components/brave_wallet_ui/components/extension/confirm-transaction-panel/index.tsx:166]: This can be moved outside of the component

**Review by josheleonard — CHANGES_REQUESTED** [review]

File: components/brave_wallet_ui/components/extension/confirm-transaction-panel/index.tsx (hunk 1 above)
Comment (Douglashdaniel) [reply components/brave_wallet_ui/components/extension/confirm-transaction-panel/index.tsx]: Good point!
Moved to our `datetime-utils` where it is also being used in our `tests` now.

**Review by josheleonard — APPROVED** [review]
LGTM

**Review by muliswilliam — APPROVED** [review]

**Comment by Douglashdaniel** [comment]
Closing this PR as `invalid` since this issue has since been addressed by the backend.

**Pull request closed without merging**